pkgs/servers/ldap/lldap/default.nix

```diff
@@ -0,0 +1,68 @@
+{ fetchFromGitHub
+, fetchzip
+, lib
+, lldap
+, nixosTests
+, rustPlatform
+}:
+
+let
+  # We cannot build the wasm frontend from source, as the
+  # wasm32-unknown-unknown rustc target isn't available in nixpkgs yet.
+  # Tracking issue: https://github.com/NixOS/nixpkgs/issues/89426
+  frontend = fetchzip {
+    url = "https://github.com/lldap/lldap/releases/download/v${lldap.version}/amd64-lldap.tar.gz";
+    hash = "sha256-/Ml4L5Gxpnmt1pLSiLNuxtzQYjTCatsVe/hE+Btl8BI=";
+    name = "lldap-frontend-${lldap.version}";
+    postFetch = ''
+      mv $out $TMPDIR/extracted
+      mv $TMPDIR/extracted/app $out
+    '';
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lldap";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "lldap";
+    repo = "lldap";
+    rev = "v${version}";
+    hash = "sha256-FAUTykFh2eGVpx6LrCjV9xWbBPH8pCgAJv3vOXFMFZ4=";
+  };
+
+  # `Cargo.lock` has git dependencies, meaning can't use `cargoHash`
+  cargoLock = {
+    # 0.4.3 has been tagged before the actual Cargo.lock bump, resulting in an inconsitent lock file.
```

Comment:
Did you raise this issue upstream already? This seems like it wouldn't affect only us?

Reply:
I hope this is a one-off error specific to the v0.4.3 release.

According to https://repology.org/project/lldap/versions, the only other package repo packaging `lldap` is the AUR.
https://aur.archlinux.org/packages/lldap is new[^1] and needs a `Cargo.lock.patch`[^2].

Which makes sense, as the AUR doesn't need to _vendor_ the whole `Cargo.lock`.
We vendor the `Cargo.lock` only because `rustPlatform.buildRustPackage` doesn't support git dependencies with `CargoHash` yet.
I thought about using a similar patch (even when I wasn't aware that it's packaged in AUR) but opted against it, as it seems somewhat redundant and bloat-y.

But to answer your question: No, this doesn't affect only us.

I intend to wait to see if this happens again with v0.4.4.
Meaning I would like to leave it how it is for now, and raise an issue only when it's still an issue in v0.4.4 :)

Will mark this as resolved for now, but feel free to comment again to mark is as unresolved again!

[^1]: https://aur.archlinux.org/cgit/aur.git/commit/?h=lldap&id=8ff0647129f810f0aacd4ace0fc8d43b69f661d5
[^2]: https://aur.archlinux.org/cgit/aur.git/diff/Cargo.lock.patch?h=lldap&id=8ff0647129f810f0aacd4ace0fc8d43b69f661d5